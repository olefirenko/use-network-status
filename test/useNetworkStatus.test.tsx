import React from 'react';
import { Demo } from '../example/Demo';
import renderer, { act } from 'react-test-renderer';

const eventListenersMap: any = {};

beforeEach(() => {
  window.addEventListener = jest.fn((event, cb) => {
    eventListenersMap[event] = cb;
  });
  window.removeEventListener = jest.fn(event => {
    eventListenersMap[event] = undefined;
  });
});
describe('useNetworkStatus', () => {
  it('renders without crashing', () => {
    const tree = renderer.create(<Demo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('set offline status correctly', () => {
    Object.defineProperty(navigator, 'onLine', { value: false });

    const tree = renderer.create(<Demo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds and remove event listeners', () => {
    let instance: renderer.ReactTestRenderer;
    act(() => {
      instance = renderer.create(<Demo />);
    });

    expect(window.addEventListener).toHaveBeenCalled();

    act(() => {
      instance.unmount();
    });

    expect(window.removeEventListener).toHaveBeenCalled();
  });
});
