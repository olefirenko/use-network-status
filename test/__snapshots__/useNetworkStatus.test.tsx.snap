// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useNetworkStatus renders without crashing 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "fontSize": "36px",
      "height": "100vh",
      "justifyContent": "center",
    }
  }
>
  ðŸ“¡ Online
</div>
`;
